src/dt2js.js

```diff
@@ -1,21 +1,60 @@
 'use strict'
 
-var yaml = require('js-yaml')
+var yap = require('yaml-ast-parser')
 var dtexp = require('datatype-expansion')
 var constants = require('./constants')
 var utils = require('./utils')
-
+var fs = require('fs')
+var path = require('path')
+var deep = require('deep-get-set')
+deep.p = true
 /**
  * Get RAML Data Types context.
  *
  * @param  {string} ramlData - RAML file content.
  * @returns  {Object} - RAML data types context.
  */
-function getRAMLContext (ramlData) {
-  var yamlContent = yaml.safeLoad(ramlData)
-  return yamlContent.types
+function getRAMLContext (ramlData, rootFileDir) {
+  rootFileDir = rootFileDir || '.'
+  var ast = yap.load(ramlData)
+  var jsContent = {}
+  traverse(jsContent, ast, rootFileDir)
+  return jsContent.types
 }
 
+function traverse (obj, ast, rootFileDir) {
+  function recurse (keys, currentNode) {
+    if (currentNode.key) {
+      keys = keys.concat([currentNode.key.value])
+    }
+    if (currentNode.value && currentNode.value.kind === 5) {
+      var filename = currentNode.value.value
+      var include
+      if (path.extname(filename) === '.json') {
```

Comment:
I would change the conditions a bit here.

From the Specs [1]:
> RAML parsers MUST parse the content of the file as RAML content and append the parsed structures to the RAML document node if the included file has a .raml, .yml, or .yaml extension or one of the following media types:
> - application/raml+yaml
> - text/yaml
> - text/x-yaml
> - application/yaml
> - application/x-yaml
> 
> Otherwise, if RAML parsers fail to parse the content and append structures, the contents of the file are included as a scalar.

So it would like this:
1) parse as RAML if: extension is .raml, .yaml, .yml OR mime types in list above
2) parse as JSON if: extension is .json OR mime type is `application/json`
3) scalar string otherwise
 
[1] https://github.com/raml-org/raml-spec/blob/master/versions/raml-10/raml-10.md#resolving-includes